sophi.php

```diff
@@ -93,6 +93,34 @@ function() {
 	}
 );
 
+
+/**
+ * Method to cleanup settings before setting the new version.
+ *
+ * @param mixed  $value     The new, unserialized option value.
+ *
+ * @return mixed
+ */
+function sophi_remove_stale_data( $value ) {
+	$is_client_id_set     = isset( $value['client_id'] );
+	$is_client_secret_set = isset( $value['client_secret'] );
+
+	if ( ! $is_client_id_set && ! $is_client_secret_set ) {
+		return $value;
+	}
+
+	if ( $is_client_id_set ) {
+		unset( $value['client_id'] );
+	}
+
+	if ( isset( $is_client_secret_set ) ) {
+		unset( $value['client_secret'] );
+	}
+
+	return $value;
+}
+add_filter( 'pre_update_option_sophi_settings', 'sophi_remove_stale_data' );
```

Comment:
I think this logic can be added to the `sanitize_settings` function in `includes/functions/settings.php` instead.